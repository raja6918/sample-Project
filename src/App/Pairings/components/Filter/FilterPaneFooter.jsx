import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ButtonMUI from '@material-ui/core/Button';

import ActionsContent from '../../../../components/Dialog/ActionsContent';

const FormFooter = styled.div`
  position: absolute;
  bottom: 0;
  width: 100%;
  text-align: right;
  background: #fff;
`;

const FilterPaneFooter = ({
  t,
  enableSave,
  handleCancel,
  // handleRefine,
  // handleAugment,
  handleApply,
}) => (
  <FormFooter>
    <ActionsContent>
      <ButtonMUI color="primary" onClick={handleCancel}>
        {t('FILTER.pane.buttons.cancel')}
      </ButtonMUI>
      {/* <ButtonMUI
        disabled={!enableSave}
        onClick={handleRefine}
        variant="contained"
        color="primary"
      >
        {t('FILTER.pane.buttons.refine')}
      </ButtonMUI>
      <ButtonMUI
        disabled={!enableSave}
        onClick={handleAugment}
        variant="contained"
        color="primary"
      >
        {t('FILTER.pane.buttons.augment')}
      </ButtonMUI> */}
      <ButtonMUI
        disabled={!enableSave}
        onClick={handleApply}
        variant="contained"
        color="primary"
      >
        {t('FILTER.pane.buttons.apply')}
      </ButtonMUI>
    </ActionsContent>
  </FormFooter>
);

FilterPaneFooter.propTypes = {
  t: PropTypes.func.isRequired,
  enableSave: PropTypes.bool.isRequired,
  handleCancel: PropTypes.func.isRequired,
  // handleRefine: PropTypes.func.isRequired,
  // handleAugment: PropTypes.func.isRequired,
  handleApply: PropTypes.func.isRequired,
};

export default FilterPaneFooter;
