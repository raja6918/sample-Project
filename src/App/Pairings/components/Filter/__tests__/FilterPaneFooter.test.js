import React from 'react';
import { shallow } from 'enzyme';
import FilterPaneFooter from '../FilterPaneFooter';

describe('Test FilterPaneFooter Component', () => {
  const minProps = {
    t: jest.fn(),
    enableSave: true,
    handleCancel: jest.fn(),
    handleApply: jest.fn(),
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<FilterPaneFooter {...minProps} />);
  });

  test('FilterPaneFooter Component renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('handleCancel should be called on Cancel button click', () => {
    wrapper
      .find({ color: 'primary' })
      .at(0)
      .simulate('click');
    expect(minProps.handleCancel).toBeCalled();
  });

  test('handleApply should be called on Apply button click', () => {
    wrapper
      .find({ color: 'primary' })
      .at(1)
      .simulate('click');
    expect(minProps.handleApply).toBeCalled();
  });
});
