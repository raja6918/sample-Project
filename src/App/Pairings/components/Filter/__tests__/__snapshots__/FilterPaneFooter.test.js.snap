// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FilterPaneFooter Component FilterPaneFooter Component renders correctly 1`] = `
<FilterPaneFooter__FormFooter>
  <ActionsContent>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[MockFunction]}
    />
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={false}
      onClick={[MockFunction]}
      variant="contained"
    />
  </ActionsContent>
</FilterPaneFooter__FormFooter>
`;
